src/nunavut/lang/cpp/templates/deserialization.j2

```diff
@@ -222,20 +222,26 @@
 {% set ref_size_bytes = 'size_bytes' |to_template_unique_name %}
 {% set ref_delimiter  = 'dh' |to_template_unique_name %}
     {
-        {{ typename_unsigned_length }} {{ ref_size_bytes }} = in_buffer.size() / 8U;
+        {{ typename_unsigned_length }} {{ ref_size_bytes }};
```

Comment:
A default initializer is needed anyway. The old construct seems fine; the optimizer will throw the extra assignment away anyway, then you can remove the else branch at the bottom.

Reply:
fixed